src/core/algorithms/gfd/gfd_miner.cpp

```diff
@@ -0,0 +1,667 @@
+                prev = premises;
+                conclusion = {l};
+            } else {
+                conclusion.push_back(l);
+            }
+        }
+        Gfd gfd = Gfd(pattern, prev, conclusion);
+        result.push_back(gfd);
+        forbidden_rules->push_back({prev, conclusion});
+    }
+}
+
+void HorizontalSpawn(graph_t const& graph, std::size_t const sigma,
+                     std::vector<graph_t> const& patterns,
+                     std::vector<Embeddings> const& embeddings_set,
+                     std::vector<Rules>& forbidden_rules_set,
+                     std::map<std::string, Info> const& attrs_info, std::vector<Gfd>& result) {
+    for (std::size_t i = 0; i < patterns.size(); ++i) {
+        auto pattern = patterns.at(i);
+        auto embeddings = embeddings_set.at(i);
+        auto forbidden_rules = std::next(forbidden_rules_set.begin(), i);
+
+        std::vector<Literal> literals = GenerateLiterals(pattern, attrs_info);
+        std::vector<SingleRule> rules =
+                GenerateRules(literals, graph, pattern, embeddings, forbidden_rules, sigma);
+        AddCompacted(rules, result, pattern, forbidden_rules);
+    }
+}
+
+class CompareCallback {
+private:
+    graph_t const& query_;
+    Embedding& iso_;
+    bool& res_;
+
+public:
+    CompareCallback(graph_t const& query_, Embedding& iso_, bool& res_)
+        : query_(query_), iso_(iso_), res_(res_) {}
+
+    template <typename CorrespondenceMap1To2, typename CorrespondenceMap2To1>
+    bool operator()(CorrespondenceMap1To2 f, CorrespondenceMap2To1) const {
+        BGL_FORALL_VERTICES_T(u, query_, graph_t) {
+            iso_.emplace(u, get(f, u));
+        }
+        res_ = true;
+        return false;
+    }
+};
+
+struct VCompare {
+    graph_t const& query;
+    graph_t const& graph;
+
+    bool operator()(vertex_t fr, vertex_t to) const {
+        return query[fr].attributes.at("label") == graph[to].attributes.at("label");
+    }
+};
+
+struct ECompare {
+    graph_t const& query;
+    graph_t const& graph;
+
+    bool operator()(edge_t fr, edge_t to) const {
+        return query[fr].label == graph[to].label;
+    }
+};
+
+void UpdateExisted(bool& isoExists, graph_t const& new_pattern,
+                   std::vector<graph_t> const& new_patterns,
+                   std::vector<Rules>& new_forbidden_rules_set, Rules const& forbidden_rules) {
+    int index = -1;
+    for (auto const& existed : new_patterns) {
+        index++;
+        VCompare vcompare{new_pattern, existed};
+        ECompare ecompare{new_pattern, existed};
+        Embedding iso;
+        CompareCallback callback(new_pattern, iso, isoExists);
+        boost::vf2_subgraph_iso(new_pattern, existed, callback,
+                                get(boost::vertex_index, new_pattern),
+                                get(boost::vertex_index, existed),
+                                vertex_order_by_mult(new_pattern), ecompare, vcompare);
+        if (isoExists) {
```

Comment:
Remove this indentation level by converting to guard clause
```suggestion
        if (!isoExists) continue;
```